catroidTest/src/org/catrobat/catroid/uitest/ui/activity/PocketMusicTest.java

```diff
@@ -72,22 +75,48 @@ public void testOrientation() throws NameNotFoundException {
 	}
 
 	public void testPianoElement() {
-		prepareTest();
 		solo.waitForActivity(PocketMusicActivity.class.getSimpleName());
 
 		assertNotNull("Piano Element was not found.", solo.getCurrentActivity().findViewById(R.id.musicdroid_piano));
 	}
 
 	public void testNoteGridElement() {
-		prepareTest();
 		solo.waitForActivity(PocketMusicActivity.class.getSimpleName());
 
 		assertNotNull("NoteGrid Element was not found.", solo.getCurrentActivity().findViewById(R.id
 				.musicdroid_note_grid));
 	}
 
+	public void testAllButtonToggle() {
+		solo.waitForActivity(PocketMusicActivity.class.getSimpleName());
+
+		TrackView trackView = (TrackView) solo.getCurrentActivity().findViewById(R.id.musicdroid_note_grid);
+		clickAndAssertButtons(trackView, true, "Button not toggled");
+		clickAndAssertButtons(trackView, false, "Button toggled");
+	}
+
+	private void clickAndAssertButtons(TrackView trackView, boolean expectedState, String assertionText) {
+		for (TrackRowView trackRowView : trackView.getTrackRowViews()) {
+			for (NoteView noteView : trackRowView.getNoteViews()) {
+				solo.clickOnView(noteView);
+				solo.sleep(200);
+				assertEquals(assertionText, noteView.isToggled(), expectedState);
+			}
+		}
+	}
```

Comment:
Thomas said on the last core-team Meeting that we didnt have to to test if buttons are pressed when we click on it. Its a standard Android functionaltity. We should not make too much UI-tests, is this test necessary?